src/Vector3.h

```diff
@@ -393,11 +393,9 @@ class VectorHasher
 	{
 		// Guaranteed to have no hash collisions for any 128x128x128 area
 		size_t Hash = 0;
-		Hash ^= static_cast<size_t>(a_Vector.x);
-		Hash <<= 8;
-		Hash ^= static_cast<size_t>(a_Vector.y);
-		Hash <<= 8;
-		Hash ^= static_cast<size_t>(a_Vector.z);
+		Hash ^= a_Vector.x + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
+		Hash ^= a_Vector.y + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
+		Hash ^= a_Vector.z + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
```

Comment:
As a diagram on a 32 bit system:
```
x: 16 15 14 13 12 11 10  9  8  7  6  5  4  3  2  1  0
y:  3  2  1  0 21 20 19 18 17 16 15 14 13 12 11 10  9  8 7  6  5  4
z: 31 30 29 28 27 26 25 24 23 22 21 20 19 18 17 16 15 14 13 12 11 10 9 8 7 6 5 4 3 2 1 0
```
The each vertical column is the bits that are xored to get a result bit. The number is which bit from that component are we shifting where. For better entropy we could take to unused high bits of the x and y component, and xor them in at the bottom of the hash.